### PR TITLE
feat(EMS-1979): Cypress optimisation

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Optimise cypress tests execution

## Resolution
* Previous job successful validation using `needs`
* Project segregation.
* Project ID as a secret.
* Concurrency introduction
* Concurrent machine id as a tag `#1` on Cypress cloud.
* Increased concurrent machines to `10`.
* Set `fail-fast` to `true`.

### CODE REVIEW

**Review by ttbarnes — DISMISSED** [review]
Beautiful

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -213,20 +220,25 @@ jobs:
...
     environment:
       name: ${{ needs.setup.outputs.environment }}
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: true
+      # Single dimension matrix - 10 jobs
       matrix:
-        machines: [1, 2, 3, 4]
+        machines: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```
Comment (Zainzzkk) [inline .github/workflows/test.yml:237]: not a blocker, quote runs really quick with 4-6 if we wanted to save resources

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -269,18 +283,25 @@ jobs:
...
+    needs: [setup, unit-tests, api-tests, e2e-tests-insurance, e2e-tests-quote]
     environment:
       name: ${{ needs.setup.outputs.environment }}
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: true
+      # Single dimension matrix - 10 jobs
+      matrix:
+        machines: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```
Comment (Zainzzkk) [inline .github/workflows/test.yml:300]: 1 machine is enough for this, it runs on 1 machine in less than 5mins usually

**Review by Zainzzkk — DISMISSED** [review]

File: e2e-tests/switch-between-services/cypress.config.js [hunk 3]
```diff
@@ -9,7 +9,6 @@ dotenv.config();
  * Cypress configuration
  */
 const cypressConfig = defineConfig({
-  projectId: 'sefntb',
```
Comment (ttbarnes) [inline e2e-tests/switch-between-services/cypress.config.js:12]: Why is `projectId` removed for all instances?

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**